partner_sync_gp/model/partner_compassion.py

```diff
@@ -78,12 +88,18 @@ def write(self, vals):
             the partner in the MySQL table used by GP. This method is also
             called by GP with XMLRPC, so that the update syncs the two
             databases. """
-        uid = self.env.user.id
-        gp = gp_connector.GPConnect()
-
         # We first call the original write method and update the records.
         result = super(ResPartner, self).write(vals)
 
+        # Don't synchronize if nothing relevant is changed.
+        for key in vals.iterkeys():
+            if key in SYNC_FIELDS:
+                break
+        else:
+            return result
```

Comment:
No it's a `for ... else`
http://psung.blogspot.ch/2007/12/for-else-in-python.html